bindings/rust/s2n-tls/src/testing.rs

```diff
@@ -325,3 +335,209 @@ pub fn poll_tls_pair_result(pair: &mut Pair<Harness, Harness>) -> Result<()> {
         }
     }
 }
+
+type LocalDataBuffer = RefCell<VecDeque<u8>>;
+
+/// CleanHarness is a testing utility used to easily test handshakes and send data.
+///
+/// SAFETY: if the server or client connection is moved outside of the struct, IO
+/// is not safe to perform. The connections use pointers to data buffers owned by
+/// the Harness. If the Harness goes out of scope, the data buffers will be dropped
+/// and the pointers will be invalid.
```

Comment:
Just curious: is there not a way to prevent the connection from being moved? Isn't this what `Pin` is for?

Reply:
No you're probably right - the added complexity probably isn't worth it for a test harness. I just wasn't sure why Pin wasn't helpful, but that makes sense!